File: solutions/java/acronym/1/src/main/java/Acronym.java
class Acronym {

    String acronym;

    Acronym(String phrase) {
        String acro="";
        char[] chPhrase = phrase.toCharArray();
        for(int i=0; i<chPhrase.length; i++){
            if(i==0 ||
               Character.isWhitespace(chPhrase[i-1]) ||
              chPhrase[i-1]=='-'|| chPhrase[i-1]=='_'){
                if(Character.isLetter(chPhrase[i]))
                    acro += chPhrase[i];
              }
        }
        acronym = acro.toUpperCase();
    }

    String get() {
        return acronym;
    }

}

File: solutions/kotlin/acronym/1/src/main/kotlin/Acronym.kt
object Acronym {
    
    fun generate(phrase: String) : String {
        var  acro = "" 
        for(i in 0..phrase.length){
            if (i==0 || phrase[i-1].isWhitespace() || 
                    setOf('-','_').contains(phrase[i-1])) {
            if (phrase[i].isLetter())
                acro += phrase[i].toUpperCase()
            }
        }
        return acro
    }
}
